modules/webapps/appservice/module.tf

```
@@ -24,6 +24,13 @@ resource "azurerm_app_service" "app_service" {
   enabled                 = lookup(var.settings, "enabled", null)
   https_only              = lookup(var.settings, "https_only", null)
 
+  key_vault_reference_identity_id = try(
```

```suggestion
  key_vault_reference_identity_id = can(var.settings.key_vault_reference_identity) ? var.combined_objects.managed_identities[try(var.settings.key_vault_reference_identity.lz_key, var.client_config.landingzone_key)][var.settings.key_vault_reference_identity.key].id : try(var.settings.key_vault_reference_identity.id, null)
```
This code is more robust and will prevent destroy/recreate. We've been transitioning to this pattern recently.
Also replaced var.settings.identity by var.settings.key_vault_reference_identity

---

Adjusted to suggested solution